libimage/push.go

```diff
@@ -63,14 +99,19 @@ func (r *Runtime) Push(ctx context.Context, source, destination string, options
 		destRef = dockerRef
 	}
 
+	// If using --all-tags, must push to registry
+	if destRef.Transport().Name() != dockerTransport.Transport.Name() && options.AllTags {
+		return nil, fmt.Errorf("--all-tags can only be used with docker transport")
+	}
+
```

Comment:
I can’t se why this function cares; if the `AllTags` implementation wants to ensure this, it should do that where `AllTags` is implemented.